src/miner_ebus.erl

```diff
@@ -46,52 +47,66 @@ init(_Args) ->
-                {ok, TxnBin} ->
-                    {reply, [{array, byte}], [TxnBin], State};
-                {'EXIT', Why} ->
-                    lager:warning("Error requesting assert loc: ~p", [Why]),
-                    {reply_error, ?MINER_ERROR_BADARGS, Member, State}
+        {ok, [OwnerB58, _Fee, _StakingFee, PayerB58]} ->
+            case
+                call_unary(Connection, add_gateway, #{
+                    owner => libp2p_crypto:b58_to_bin(OwnerB58),
+                    payer => libp2p_crypto:b58_to_bin(PayerB58),
+                    staking_mode => full
+                })
+            of
+                {ok, #{result := #{add_gateway_txn := BinTxn}}} ->
+                    {reply, [{array, byte}], [BinTxn], State};
+                {error, Error} ->
+                    lager:warning("Error requesting add gateway from embedded gateway api: ~p", [Error]),
+                    {reply_error, ?MINER_ERROR_INTERNAL, Member, State#state{ connection = undefined }}
             end;
         {ok, Args} ->
             lager:warning("Invalid add gateway args: ~p", [Args]),
             {reply_error, ?MINER_ERROR_BADARGS, Member, State};
         {error, Error} ->
             lager:warning("Invalid add gateway args: ~p", [Error]),
-            {reply_error, ?MINER_ERROR_BADARGS, Member, State}
+            {reply_error, ?MINER_ERROR_BADARGS, Member, State#state{ connection = undefined }}
     end;
-handle_message(?MINER_OBJECT(?MINER_MEMBER_ASSERT_LOC)=Member, Msg, State=#state{}) ->
-    case ebus_message:args(Msg) of
-        {ok, [H3String, OwnerB58, Nonce, StakingFee, Fee, PayerB58]} ->
-            lager:info("Requesting assert for ~p", [H3String]),
-            case (catch blockchain:assert_loc_txn(H3String, OwnerB58, PayerB58, Nonce, StakingFee, Fee)) of
-                {ok, TxnBin} ->
-                    {reply, [{array, byte}], [TxnBin], State};
-                {'EXIT', Why} ->
-                    lager:warning("Error requesting assert loc: ~p", [Why]),
-                    {reply_error, ?MINER_ERROR_BADARGS, Member, State}
-            end;
-        {ok, Args} ->
-            lager:warning("Invalid asset_loc args: ~p", [Args]),
-            {reply_error, ?MINER_ERROR_BADARGS, Member, State};
-        {error, Error} ->
-            lager:warning("Invalid assert_loc args: ~p", [Error]),
-            {reply_error, ?MINER_ERROR_BADARGS, Member, State}
+handle_message(?MINER_OBJECT(?MINER_MEMBER_ASSERT_LOC) = Member, _Msg, State) ->
+    lager:warning("Assert_loc deprecated; miners now obtain loc data on-chain", []),
+    {reply_error, ?MINER_ERROR_BADARGS, Member, State};
+handle_message(?MINER_OBJECT(?MINER_MEMBER_P2P_STATUS) = Member, _, State=#state{ connection = Connection }) ->
+    case call_unary(Connection, height, #{}) of
+        {ok, #{result := #{height := Height}}} ->
+            {reply, [{array, {struct, [string, string]}}], [{"connected", "yes"}, {"height", integer_to_list(Height)}], State};
```

Comment:
this, and gateway_config's notion of connected may change if we do what @andymck is working towards which is throwing out the hotspot connection if it doesn't have an asserted location